damnit/api.py

```diff
@@ -171,6 +173,110 @@ def summary(self):
                 return blob2complex(value)
             return value
 
+    def preview_data(self, *, data_fallback=True, deserialize_plotly=True):
+        """Get the preview data for the variable
+
+        May return a 1D or 2D data array, a 3D RGB(A) arrray, a plotly figure
+        object, a str of plotly JSON (with deserialize_plotly=False) or None
+        if no preview is available.
+
+        If no preview was specified in the context file, but the returned data
+        meets the conditions for a preview, this can be returned instead.
+        Pass `data_fallback=False` to prevent this.
+        """
+        with h5py.File(self._h5_path) as f:
+            xarray_group = dset = None
+            if (obj := f.get(f".preview/{self.name}")) is not None:
+                # Explicit preview
+                type_hint = self._type_hint(obj)
+                if isinstance(obj, h5py.Group):
+                    xarray_group = obj.name
+                else:
+                    dset = obj
+            elif data_fallback:
+                # Implicit: use data as preview if suitable
+                grp = f[self.name]
+                type_hint = self._type_hint(grp)
+                if type_hint is DataType.DataArray:
+                    xarray_group = self.name
+                elif type_hint is DataType.Dataset:
+                    return None
+                else:
+                    dset = grp['data']
+            else:
+                return None
+
+            if xarray_group is not None:
+                for obj in f[xarray_group].values():
+                    if isinstance(obj, h5py.Dataset) and (
+                        obj.ndim > 3 or (obj.ndim == 3 and obj.shape[-1] not in (3, 4))
+                    ):
```

Comment:
A Dataset doesn't have `ndim` or `shape` properties, it's just a container for other arrays. We could perhaps return a string like the summary instead? Though that's not really a preview.

```suggestion
                    if isinstance(obj, h5py.Dataset):
```

Reply:
Oh NVM, I misread this as an `xr.Dataset` rather than an `h5py.Dataset` :upside_down_face: